### ISSUE TITLE
"no available server" when connect to a container using definition of IngressRouteTCP. Restart helps. 2.2.8

### ISSUE THREAD

**Issue by kopaygorodsky** [traefik/traefik#7167]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Traefik 2.2.8. I create IngressRouteTCP with tls passthrough enabled.  In a few minutes I try to connect to the container and my app says "context deadline exceeded" and in traefik logs "no available server". On the traefik dashboard everything is ok.  Problem is solved once I restart traefik pod. 
Works fine on 2.0.1
```yml
- apiVersion: traefik.containo.us/v1alpha1
  kind: IngressRouteTCP
  metadata:
    creationTimestamp: "2020-08-05T19:55:41Z"
    generation: 1
    labels:
      node: a2
      resourceName: org1peers
    name: peerset-org1peers-a4-org1peers
    namespace: default
    ownerReferences:
    - apiVersion: peerset.x/v1beta
      controller: true
      kind: PeerSet
      name: org1peers
      uid: 3f91c80c-ebd6-4a1c-9d64-b0de29894853
    resourceVersion: "55267"
    selfLink: /apis/traefik.containo.us/v1alpha1/namespaces/default/ingressroutetcps/peerset-org1peers-a2-org1peers
    uid: d6145fa0-4560-476a-8e74-98d7bb90f1bd
  spec:
    entryPoints:
    - websecure
    routes:
    - match: HostSNI(`peerset-org1peers-a4-org1peers.proxy.bank1.x.com`)
      services:
      - name: peerset-org1peers-a4-org1peers
        port: 9111
    tls:
      passthrough: true

---

apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2020-08-05T19:55:34Z"
  labels:
    app: proxy
    node: a1
    resourceName: org1peers
  name: peerset-org1peers-a4-org1peers
  namespace: default
  ownerReferences:
  - apiVersion: peerset.x/v1beta
    controller: true
    kind: PeerSet
    name: org1peers
    uid: 3f91c80c-ebd6-4a1c-9d64-b0de29894853
  resourceVersion: "55194"
  selfLink: /api/v1/namespaces/default/services/peerset-org1peers-a1-org1peers
  uid: fc68509b-e867-46d3-be3a-eb262507b2d6
spec:
  externalName: peerset-org1peers-a4.org1peers.default.svc.cluster.local #<------ a pod of some StatefulSet like controller (have crd with similar way of handling dns for a container using obj.ServiceName and pod.Host)
  ports:
  - name: peerset-org1peers-a4
    port: 9111
    protocol: TCP
    targetPort: 9111
  sessionAffinity: None
  type: ExternalName
status:
  loadBalancer: {}
```

### What did you expect to see?

I expected to be connected to the pod in a few seconds after IngressRouteTCP is created with no restart of the traefik pod.

### What did you see instead?

context deadline exceeded 

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.2.8
on 2.0.1 works fine 
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
Kubernetes GKE 1.16.13-gke.1
Default installation of traefik using helm `helm install traefik2 traefik/traefik --namespace kube-system`
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)
After creation of IngressRouteTCP there is an error 
```
In service \"default-peerset-org1peers-a4-org1peers-425aaf203a1844d04c5f@kubernetescrd\" server \"peerset-org1peers-a4.org1peers.default.svc.cluster.local:9111\": lookup peerset-org1peers-a4.org1peers.default.svc.cluster.local on 10.52.0.10:53: no such host
```
It's because of service defined in ingress has `ExternalName` type and it points to a pod of a StatefulSet which could being creating at the moment. In a few seconds `nslookup peerset-org1peers-a4.org1peers.default.svc.cluster.local` returns host inside of the traefik container. 
<img width="623" alt="image" src="https://user-images.githubusercontent.com/2415637/90120130-17dbc100-dd63-11ea-8c0d-55cfb2991f6b.png">

When I connect -> 
`" level=error msg="no available server"`
```
(paste your output here)
```

**Comment by kopaygorodsky** [comment]
Just installed 2.0.1 again and it works fine.

**Comment by kopaygorodsky** [comment]
I'm not sure this is a questions, it should be a bug.

**Comment by ldez** [comment]
Hello,

the current maintained version of Traefik is v2.2: https://github.com/containous/traefik#release-cycle

Could try with the latest version (v2.2.8), thanks.

**Comment by kopaygorodsky** [comment]
I tried 2.2.8 and it does not work, but on 2.0.1 it does.

**Comment by SantoDE** [comment]
Does it only happen when the service referenced in the `IngressRouteTCP` is an external service?

**Comment by kopaygorodsky** [comment]
I think yes, because in types of service there is always assigned ip, so "no such host" will never appear if I'm correct. 
I don't have a chance to test it because all my grpc containers are under headless service and I use ExternalName only to be able to specify service name in ingress definition since podName.headlessServiceName.namespace.cluster.local is not supported as an endpoint.

**Comment by SantoDE** [comment]
Hey @kopaygorodsky ,

I'm afraid I can't really reproduce the issue. It might be something with your cluster or so? The only option is, that you can provide us with a reproducible example I'm afraid :(

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of tidying up the issues.

I have to close this one because of its lack of activity :disappointed:

Feel free to re-open it or join our [Community Forum](https://community.containo.us/).

**Comment by kopaygorodsky** [comment]
I checked on fresh new clusters created on EKS 1.16. Works very randomly. I posted my research regarding this issue under #5707. Seems it could be the problem.

**Comment by kopaygorodsky** [comment]
deleting any pod that is exposed by traefik, triggers reload of the config(I see it in logs) and then error is gone

**Comment by kopaygorodsky** [comment]
According to debug logs `In service \"default-peerset-org1peers-a4-org1peers-425aaf203a1844d04c5f@kubernetescrd\" server \"peerset-org1peers-a4.org1peers.default.svc.cluster.local:9111\": lookup peerset-org1peers-a4.org1peers.default.svc.cluster.local on 10.52.0.10:53: no such host` appears just when I created new IngressRouteTCP

and tcp proxy is not created
https://github.com/containous/traefik/blob/8b8b1427f678c2a8b0a759da377ca84c9522f0f6/pkg/server/service/tcp/service.go#L62

**Comment by kopaygorodsky** [comment]
I managed to fix my k8s operator so it makes sure that pod is created before ingress. `no service available` does not appear anymore, but I run into the same issue when I deleted the pod :`Error while connection to backend: dial tcp 10.56.10.76:9111: connect: no route to host`. The thing is after restart Endpoints config is updated with new IP of pod, but traefik continues to connect to the old one.

**Comment by SantoDE** [comment]
Hey 👋 

We've reopened the issue, but we still need a reproducible example. Can you provide us with one, please?

**Comment by kopaygorodsky** [comment]
https://gist.github.com/kopaygorodsky/5652bfdd255771e3eef511aeede0205f

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of tidying up the issues.

I have to close this one because of its lack of activity :disappointed:

Feel free to re-open it or join our [Community Forum](https://community.containo.us/).

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of tidying up the issues.

I have to close this one because of its lack of activity :disappointed:

Feel free to re-open it or join our [Community Forum](https://community.containo.us/).

**Comment by traefiker** [comment]
Closed by #7370.
